source/validate.cpp

```diff
@@ -117,6 +119,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Parses OpExtension instruction and registers extension.
+void RegisterExtension(ValidationState_t& _,
+                       const spv_parsed_instruction_t* inst) {
+  const std::string extension_str = libspirv::GetExtensionString(inst);
+  Extension extension;
+  if (!ParseSpvExtensionFromString(extension_str, &extension)) {
+    _.diag(SPV_SUCCESS) << "Failed to parse OpExtension " << extension_str;
+    return;
+  }
+
+  _.RegisterExtension(extension);
+}
+
+spv_result_t ProcessExtensions(
+    void* user_data, const spv_parsed_instruction_t* inst) {
+  const SpvOp opcode = static_cast<SpvOp>(inst->opcode);
+  if (opcode == SpvOpCapability)
```

Comment:
It's very much worth saying how this fits with the parser:

For a valid module, the intial segment of instructions are just optional OpCapability followed by OpExtension. Terminate parsing early when you get past those.